src/ui/component/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,49 @@
+import type { FC } from 'react'
+import { useCallback } from 'react'
+import classNames from 'classnames'
+import * as RCheckbox from '@radix-ui/react-checkbox'
+import { Icon } from '@/ui/component/icon/icon'
+import './checkbox.scss'
+
+export type CheckboxOption = {
+  id: string
+  name: string
+  checked: boolean
+  disabled: boolean
+}
+
+type CheckboxProps = {
+  id: string
+  name: string
+  checked: boolean
+  disabled: boolean
+  onCheckedChange: (checkboxAttributes: CheckboxOption) => void
+}
+
+export const Checkbox: FC<CheckboxProps> = ({ id, name, checked, disabled, onCheckedChange }) => {
+  const handleCheckedChange = useCallback(() => {
+    if (disabled) return
+
+    onCheckedChange({
+      checked: !checked,
+      id,
+      name,
+      disabled
```

Comment:
Not sur you need to return disabled here, because you wrote:
`if (disabled) return ` juste above, so It's automatically `false` here

Reply:
Yes you are right, I kept it in order to pass the entire Option, making it easier to deal with. I can remove it, but we need to pass `Pick<CheckboxOption, 'checked' | 'id' | 'name'>`, which is probably fine. What do you think ?

Reply:
I think that if we don't need a property, we should not pass it :)